### PR TITLE
add blog post for Flagger release

### PR DESCRIPTION
**Pull request by dholbach**
To coincide with Flagger 1.19.0 release.

Only merge after the Flagger release and checking that https://docs.flagger.app/tutorials/gatewayapi-progressive-delivery is online.

Signed-off-by: Daniel Holbach <daniel@weave.works>

### CODE REVIEW

File: content/en/blog/2022-03-11-flagger-is-adding-gateway-api/index.md [hunk 1]
```diff
@@ -0,0 +1,213 @@
...
+description: New Flux and Flagger releases bring more security, many new adopters have joined our community, Flux articles and docs, upcoming Flux events helping you get started and more.
+url: /blog/2022/03/flagger-adds-gateway-api-support/
+resources:
+- src: "**.png"
+  title: "Image #:counter"
+tags: [flagger]
+---
+
+The Flagger team is proud to bring you Kubernetes Gateway API support as
+part of the 1.19.0 release. Read here about why this is a significant
```
Comment (dholbach) [inline content/en/blog/2022-03-11-flagger-is-adding-gateway-api/index.md:14]: Add link

File: content/en/blog/2022-03-11-flagger-is-adding-gateway-api/index.md [hunk 2]
```diff
@@ -0,0 +1,213 @@
...
+
+Flagger creates a `HTTPRoute` with the appropriate header matches, path
+matches, etc and attaches the primary and canary service to the
+`HTTPRoute`. During the canary analysis, the weights related to both the
+services are adjusted accordingly.
+
+![Flagger using Gateway API](featured-flagger-gatewayapi-canary.png)
+
+If you want to get started right away, have a look at [our
+tutorial](https://github.com/aryan9600/flagger/blob/aryan9600/gateway-api/docs/gitbook/tutorials/gatewayapi-progressive-delivery.md),
```
Comment (dholbach) [inline content/en/blog/2022-03-11-flagger-is-adding-gateway-api/index.md:105]: Update link

File: content/en/blog/2022-03-11-flagger-is-adding-gateway-api/index.md [hunk 3]
```diff
@@ -0,0 +1,213 @@
...
+>   This promotes a highly portable core API (like Ingress) that still
+>   gives flexibility for Gateway controller implementers.
+
+![Gateway API Model](api-model.png)
+
+Gateway API exposes a more general API than Ingress for proxying and you
+can use it for more protocols than just HTTP (although most
+implementations support just HTTP for now). It models more
+infrastructure components to provide better deployment and management
+options. There are three core components to the Gateway API:
```
Comment (dholbach) [inline content/en/blog/2022-03-11-flagger-is-adding-gateway-api/index.md:48]: @stefanprodan Can you review this section as well?

**Comment by dholbach** [comment]
→ https://deploy-preview-827--fluxcd.netlify.app/blog/

Thanks @aryan9600 for all your help!

File: content/en/blog/2022-03-11-flagger-is-adding-gateway-api/index.md [hunk 4]
```diff
@@ -0,0 +1,213 @@
...
+this integration is.
+
+## Flagger works with all implementations
+
+With added support for Gateway API, Flagger now works with all
+[implementations](https://gateway-api.sigs.k8s.io/implementations/),
+which means that as of today these are natively supported: Apache
+APISIX, Contour, Emissary-Ingress, Gloo Edge 2.0, Google Kubernetes
+Engine, HAProxy Ingress, HashiCorp Consul, Istio, Kong and Traefik.
+
```
Comment (stefanprodan) [inline content/en/blog/2022-03-11-flagger-is-adding-gateway-api/index.md:118]: The Flagger team has successfully tested Contour and Istio using the v1beta2 Gateway API.  Starting with Flagger v1.19, the Gateway API is part of our end-to-end test suite using the Contour implementation.

File: content/en/blog/2022-03-11-flagger-is-adding-gateway-api/index.md [hunk 5]
```diff
@@ -0,0 +1,213 @@
...
+Flagger creates a `HTTPRoute` with the appropriate header matches, path
+matches, etc and attaches the primary and canary service to the
+`HTTPRoute`. During the canary analysis, the weights related to both the
+services are adjusted accordingly.
+
+![Flagger using Gateway API](featured-flagger-gatewayapi-canary.png)
+
+If you want to get started right away, have a look at [our
+tutorial](https://docs.flagger.app/tutorials/gatewayapi-progressive-delivery),
+which shows you how to use Gateway API and Flagger to automate canary
```
Comment (stefanprodan) [inline content/en/blog/2022-03-11-flagger-is-adding-gateway-api/index.md:107]:
```suggestion
which shows you how to use Contour's Gateway API implementation and Flagger to automate canary
```

File: content/en/blog/2022-03-11-flagger-is-adding-gateway-api/index.md [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+  Route resources to evaluate for routing, amongst other things.
+- `HTTPRoute`: This is a `Route` resource that is specific for HTTP
+  requests. It defines routing rules such as filters, path and
+  header matches, etc. and which services should the request be
+  forwarded to.
+
+## How does this work in Flagger?
+
+Flagger makes use of the fact that `HTTPRoute` allows users to define a
+weight related to each routing rule. These weights are used to determine
```
Comment (aryan9600) [inline content/en/blog/2022-03-11-flagger-is-adding-gateway-api/index.md:65]:
```suggestion
weight related to each reference to a service inside a routing rule. These weights are used to determine
```

File: content/en/blog/2022-03-11-flagger-is-adding-gateway-api/index.md [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+tutorial](https://docs.flagger.app/tutorials/gatewayapi-progressive-delivery),
+which shows you how to use Contour's Gateway API implementation and Flagger to automate canary
+deployments. It won't take long to follow, but will convey how powerful
+this integration is.
+
+## Flagger works with all implementations
+
+With added support for Gateway API, Flagger now works with all
+[implementations](https://gateway-api.sigs.k8s.io/implementations/),
+which means that as of today these are natively supported: Apache
```
Comment (stefanprodan) [inline content/en/blog/2022-03-11-flagger-is-adding-gateway-api/index.md:115]: Apache APISIX and Gloo are WIP, please remove them.

File: content/en/blog/2022-03-11-flagger-is-adding-gateway-api/index.md [hunk 8]
```diff
@@ -0,0 +1,217 @@
...
+  - name: example-gateway
+  hostnames:
+  - "foo.example.com"
+  rules:
+  - matches:
+    - path:
+        type: PathPrefix
+        value: /login
+    backendRefs:
+    - name: foo-svc
```
Comment (stefanprodan) [inline content/en/blog/2022-03-11-flagger-is-adding-gateway-api/index.md:85]:
```suggestion
    - name: foo-primary
```

File: content/en/blog/2022-03-11-flagger-is-adding-gateway-api/index.md [hunk 9]
```diff
@@ -15,6 +15,21 @@ part of the [1.19.0 release](https://github.com/fluxcd/flagger/releases/tag/v1.1
 Read here about why this is a significant development in Flagger and how
 you can make use of it.
 
+## What is Flagger?
+
+Flagger is a progressive delivery tool that automates the release process for applications running on Kubernetes.
```
Comment (dholbach) [inline content/en/blog/2022-03-11-flagger-is-adding-gateway-api/index.md:20]: Maybe add a link to the Flagger website?

**Review by dholbach — COMMENTED** [review]
Thanks for the edits @stefanprodan - this looks great now.

File: content/en/blog/2022-03-11-flagger-is-adding-gateway-api/index.md (hunk 9 above)
Comment (stefanprodan) [reply content/en/blog/2022-03-11-flagger-is-adding-gateway-api/index.md:20]: I added below a link to the repo.

**Review by stefanprodan — APPROVED** [review]
🚀

**Pull request merged**